src/Combinatorics/EnumerativeCombinatorics/types.jl

```diff
@@ -8,6 +8,10 @@ struct Partition{T<:IntegerUnion} <: AbstractVector{T}
   p::Vector{T}
 end
 
+struct PartitionSet{T<:IntegerUnion}
```

Comment:
Maybe this should be called `PartitionIterator`? But not really important. We have to see whether one can use the same type for the restricted partitions (with potentially unused fields) or whether there need to be several types.

Reply:
I renamed to `Partitions` because I guess it is not really a set; I don't really like including `Iterator` in the type name, but if this is the OSCAR standard for iterator types I am okay with it and can make that change.

Reply:
Sounds good to me. I'm not aware of any OSCAR standard regarding names for iterators.